superset/views/sql_lab.py

```diff
@@ -228,6 +229,29 @@ def migrate_query(  # pylint: disable=no-self-use
     def delete_query(  # pylint: disable=no-self-use
         self, tab_state_id: int, client_id: str
     ) -> FlaskResponse:
+        # Before deleting the query, ensure it's not tied to any
+        # active tab as the last query. If so, replace the query
+        # with the latest one created in that tab
+        tab_state_query = db.session.query(TabState).filter_by(
```

Comment:
We could also set the `latest_query_id` to NULL, the same as `saved_query_id`, if we don't care about reassign the latest query.